portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,220 @@
+const FAVORITE_NO_MATCH = 'No match here. Try another search or add this to your favorites!';
+// Message prompting users to click on one of the favorite links
+const FAVORITE_SEARCH_TIP = 'Tip: These are some of my favorites. Clicking on one'
+                             + ' will favorite it for you too!'
+// ID of the list containing all the search results from favorite search
+const FAVORITE_SEARCH_RESULT_LIST_ID = 'fav-search-list';
+// ID of the sidebar where user-selected favorites go
+const USER_FAVORITE_LIST_ID = 'fav-list';
+// ID of the placeholder text in the favorites are when there are no favorites selected
+const FAVORITE_PLACEHOLDER_ID = 'fav-placeholder';
+// Text in the favorite placeholder
+const FAVORITE_PLACEHOLDER_TEXT = `You haven't favorited any items yet!`;
+// Tag for creating a single checkbox in HTML
+const HTML_CHECKBOX_TAG = '<input type="checkbox"/>'
+// Property for what the user typed/selected in an input field
+const TEXT_SELECTION = 'value';
+// Property for whether a checkbox is checked or not
+const CHECKBOX_IS_CHECKED = 'checked';
+
+// dictionary of all the facts that can be pulled up
+const FACTS = {
+  movies: ['V for Vendetta', 'Monty Python and the Holy Grail',
+    'Lord of the Rings (all of them)'],
+  shows: ['Seinfeld', 'Star Trek: DS9', 'The Mandalorian', 'Night on Earth'],
+  food: ['Sushi', 'Pizza', 'Yogurt', 'Bacon', 'Lasagna', '50¢ Walmart Pies'],
+  videoGames: ['Civ 5', 'Portal 2', 'Kerbal Space Program', 'Skyrim'],
+}
+// array of all the facts, pulled from above
+const FACTS_NAMES = mergeLists();
+// lowercase array of the facts
+const FACTS_NAMES_LOWERCASE = FACTS_NAMES.map(v => v.toLowerCase());
+
+// store what the user favorites
+let userFavorites = new Set();
+
+// perform necessary setup
+document.addEventListener('DOMContentLoaded', () => {
+  // Populate from built-in favorites automatically when typing
+  document.getElementById(USER_FAVORITE_INPUT_ID).addEventListener('keyup', e => {
+    checkAgree();
+
+    if (e.key == ENTER_CODE) {
+      processNewFavorite();
+    }
+  });
+});
+
+/* Returns a (pseudo)random element of the given array */
+function randomElement(arr) {
+  return arr[Math.floor(Math.random() * arr.length)];
+}
+
+/* Merge all the lists in the dictionary facts into one list */
+function mergeLists() {
+  let out = [];
+
+  Object.keys(FACTS).forEach(key => {
+    out = out.concat(FACTS[key]);
+  });
+  return out;
+}
+
+/* Finds all occurances of substring in the values of arr and returns a list containing
+ * the full strings in arr. Not that this ignores case for searching purposes */
+function findSubstrings(arr, substring) {
+  let out = [];
+
+  substring = substring.toLowerCase();
+  arr.forEach(value => {
+    if (value.toLowerCase().includes(substring)) {
+      out.push(value);
+    }
+  });
+
+  return out;
+}
+
+// Appends a new HTML element to the given parent ID with the given information
```

Comment:
"/*...*/" is more commonly used for explaining the whole function. Please make the format of all function comments consistent.